tests/chainer_tests/functions_tests/array_tests/test_as_strided.py

```diff
@@ -0,0 +1,305 @@
+import unittest
+
+from chainer import testing, Variable, grad, cuda
+
+import numpy as np
+
+from chainer.functions import as_strided
+from chainer.functions.array.as_strided import _stride_array
+
+
+@testing.parameterize(
+    {'dtype': np.float16},
+    {'dtype': np.float32},
+    {'dtype': np.float64},
+    {'dtype': np.int16},
+    {'dtype': np.int32},
+    {'dtype': np.int64}
+)
+class TestStrideArray(unittest.TestCase):
+    def check_flip(self, xp):
+        x = xp.arange(4, dtype=self.dtype)
+        y = _stride_array(x, (4,), (-1,), 3)  # [3, 2, 1, 0]
+        y_expected = x[::-1]
+        testing.assert_allclose(y, y_expected)
+
+    def test_flip_cpu(self):
+        self.check_flip(np)
+
+    @testing.attr.gpu
+    def test_flip_gpu(self):
+        self.check_flip(cuda.cupy)
+
+    def check_broadcast(self, xp):
+        x = xp.arange(12, dtype=self.dtype).reshape((3, 4)).copy()
+        # [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]]
+        y = _stride_array(x, (2, 3, 4), (0, 4, 1), 0)
+        y_expected = xp.broadcast_to(x, (2, 3, 4))
```

Comment:
Please avoid calling `broadcast_to` for NumPy 1.9.
```
def broadcast_to(xp, x, shape):
    if hasattr(xp, 'broadcast_to'):
        return xp.broadcast_to(x, shape)
    else:
        # numpy 1.9 doesn't support broadcast_to method
        dummy = xp.empty(shape)
        return xp.broadcast_arrays(x, dummy)[0]
```

Reply:
fixed